src/Simulator/IncrementalRedstoneSimulator/DaylightSensorHandler.h

```diff
@@ -0,0 +1,65 @@
+
+#pragma once
+
+
+
+
+
+namespace DaylightSensorHandler
+{
+	inline unsigned char GetPowerLevel(const cChunk & Chunk, const Vector3i Position)
+	{
+		double PI = 3.14159;
+		int TimeOfDay = Chunk.GetWorld()->GetTimeOfDay();
+		bool IsInverted = (Chunk.GetBlock(Position) == E_BLOCK_DAYLIGHT_SENSOR) ? 0 : 1;
+		int Output = Clamp(FloorC(14 * std::sin((TimeOfDay * PI) / 12000) + 8), 0, 15);
+
+		if (IsInverted)
+		{
+			Output = 15 - Output;
+		}
+
+		return static_cast<unsigned char>(Output);
+	}
+
+	inline PowerLevel GetPowerDeliveredToPosition(const cChunk & a_Chunk, Vector3i a_Position, BLOCKTYPE a_BlockType, Vector3i a_QueryPosition, BLOCKTYPE a_QueryBlockType, bool IsLinked)
+	{
+		UNUSED(a_Chunk);
+		UNUSED(a_BlockType);
+		UNUSED(a_QueryPosition);
+
+		// Daylight sensors output to all surrounding blocks
+		// Retrieve and return the cached power calculated by Update for performance
+		return IsLinked ? 0 : DataForChunk(a_Chunk).GetCachedPowerData(a_Position);
+	}
+
+	inline void Update(cChunk & a_Chunk, cChunk & CurrentlyTicking, Vector3i a_Position, BLOCKTYPE a_BlockType, NIBBLETYPE a_Meta, const PowerLevel Power)
+	{
+		// LOGD("Evaluating Darryl the daylight sensor (%d %d %d)", a_Position.x, a_Position.y, a_Position.z);
```

Comment:
Some debug output left here

Reply:
Thanks for the tips I'll fix those. The debug output was left there b/c it's in the other redstone handlers as well but I can remove it

Reply:
Yeah - if it's everywhere we can keep the debug output